modules/pandas/derived_variables/paidhb.py

```diff
@@ -84,54 +88,73 @@ def hb_earnings_disregard(dataframe):
+    single_parent_age_16_17 = (dataframe["AGE1"] >= 16) & (dataframe["AGE1"] <= 17) & (dataframe["TOTCHILD"] > 0)
 
-    # Single Parent aged 16-17
-    dataframe.loc[(dataframe["AGE1"] >= 16) & (dataframe["AGE1"] <= 17) & (dataframe["TOTCHILD"] > 0), ["personal_allowance"]] = PERSONAL_ALLOWANCE_TYPE1
+    set_personal_allowance(dataframe, single_adult_under_25 | single_parent_age_16_17, PERSONAL_ALLOWANCE_TYPE1)
 
-    # Single adult over 25
-    dataframe.loc[(dataframe["AGE1"] >= 25) & (dataframe["TOTADULT"] == 1) & (dataframe["TOTCHILD"] == 0), ["personal_allowance"]] = PERSONAL_ALLOWANCE_TYPE2
+    single_adult_over_25 = (dataframe["AGE1"] >= 25) & (dataframe["TOTADULT"] == 1) & (dataframe["TOTCHILD"] == 0)
+    single_parent_over_18 = (dataframe["AGE1"] >= 18) & (dataframe["TOTCHILD"] > 0)
 
-    # Single Parent aged 18+
-    dataframe.loc[(dataframe["AGE1"] >= 18) & (dataframe["TOTCHILD"] > 1), ["personal_allowance"]] = PERSONAL_ALLOWANCE_TYPE2
+    set_personal_allowance(dataframe, single_adult_over_25 | single_parent_over_18, PERSONAL_ALLOWANCE_TYPE2)
 
-    # Couple, both under 18
-    dataframe.loc[ \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE2"] < 18) & (dataframe["RELAT2"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE3"] < 18) & (dataframe["RELAT3"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE4"] < 18) & (dataframe["RELAT4"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE5"] < 18) & (dataframe["RELAT5"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE6"] < 18) & (dataframe["RELAT6"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE7"] < 18) & (dataframe["RELAT7"] == "P")) | \
-        ((dataframe["AGE1"] < 18) & (dataframe["AGE8"] < 18) & (dataframe["RELAT8"] == "P")), ["personal_allowance"]] = PERSONAL_ALLOWANCE_TYPE3
-
-    # Couple, at least one over 18
-    dataframe.loc[ \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE2"] > 18)) & (dataframe["RELAT2"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE3"] > 18)) & (dataframe["RELAT3"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE4"] > 18)) & (dataframe["RELAT4"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE5"] > 18)) & (dataframe["RELAT5"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE6"] > 18)) & (dataframe["RELAT6"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE7"] > 18)) & (dataframe["RELAT7"] == "P")) | \
-        (((dataframe["AGE1"] > 18) | (dataframe["AGE8"] > 18)) & (dataframe["RELAT8"] == "P")), ["personal_allowance"]] = PERSONAL_ALLOWANCE_TYPE4
+    possible_young_couples = [couple_and_both_are_under_18(dataframe, tenant_no) for tenant_no in range(2, 9)]
+
+    young_couple = possible_young_couples[0]
+
+    for possible_couple in possible_young_couples[1:]:
```

Comment:
This feels less idiomatic to me and a bit harder to reason about. Are we comparing two rows here?